fehler-macros/src/throws.rs

```diff
@@ -141,10 +141,24 @@ fn is_unit_fn(i: &syn::ReturnType) -> bool {
     }
 }
 
-fn ok(expr: &syn::Expr) -> syn::Expr {
-    syn::parse2(quote::quote!(<_ as ::fehler::__internal::_Succeed>::from_ok(#expr))).unwrap()
+fn ok(expr: &syn::Expr, is_tail: bool) -> syn::Expr {
+    if is_tail {
+        syn::parse_quote! {
+            #[allow(unreachable_code)]
+            <_ as ::fehler::__internal::_Succeed>::from_ok(#expr)
```

Comment:
It's not great that this ends up suppressing legitimate unreachable_code lints inside the user's tail expr (which might be a big complicated `match` or something). Is there a different way?

Reply:
Adding here is certainly wrong (I was confused with the case of unit return).
It seems to be desirable to add the attribute only on an implicit unit, and inherit the original attributes when an explicit tail expr is provided.